### PR TITLE
[core] Prepare migration with main repository, eslint v1

### PR DESCRIPTION
**Pull request by oliviertassinari**
This is a smaller chunk of #1930. I have figured that it would be much easier to have this effort go into smaller efforts.

### CODE REVIEW

**Comment by dmtrKovalenko** [comment]
What the motivation of bringing this 2 parts in different PRs? I think they should never be merged independently. 

Maybe change the base branch to #1930?

File: docs/_shared/Example.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ function Example({ source, testId, paddingBottom }: ExampleProps) {
   );
 
   return (
-    <>
+    <React.Fragment>
```
Comment (dmtrKovalenko) [inline docs/_shared/Example.tsx:96]: What the motivation?

**Comment by oliviertassinari** [comment]
> What the motivation of bringing this 2 parts in different PRs? I think they should never be merged independently.

I was planning on making more than 2 pull requests, more like a couple. The incentive is incremental changes: avoid conflicts on large pull requests, makes review easier. #1930 will finalize the effort, putting constraints in place so it holds in the long term. Step by step.

File: docs/_shared/Example.tsx (hunk 1 above)
Comment (oliviertassinari) [reply docs/_shared/Example.tsx:96]: Why are we using `<React.Framgent>` in the codebase? I think that it's between these lines:

- Only this approach allows setting a `key`
- Only this approach allows new developers to easily search on Google what the element for. 
- We enforce one way for consistency. 

If you think that we should change the tradeoff, please open an RFC on the main repository to make the point. The objective here is to share the same eslint configuration on the whole codebase.

**Review by dmtrKovalenko — APPROVED** [review]

**Pull request merged**